.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,31 @@
+name: codeql
+
+on:
+  push:
+    branches:
+    - master
+    - v1.9
+    - v1.8
+    - v1.7
+  pull_request:
+    branches:
+    - master
+  schedule:
+    - cron: "6 45 * * 3"
```

Comment:
@twpayne Was there a particular reason for wanting to run this both on a schedule *and* on every push to master?

Reply:
Yes, the CodeQL service gets updated from time to time as they add more checks, so we want to take advantage of those.

Reply:
It feels a bit excessive maybe. This job takes between 17 and 25min to execute, and during that time, block the slot for another job. Would it be enough to run it on all PRs + once a day, but not on every commit?

Maybe I'm underestimating how many bugs it found so far? I also had the impression it's a static analysis tool, so running it on every single commit doesn't give us more information than running on a schedule because it's already clear what is broken when the test fails. Am I missing something?

Reply:
I don't think you're missing anything, your analysis sounds very reasonable.

It found a few potential bugs in the initial analysis, but few since. It's nice to get feedback early while the PR is still being reviewed as this allows the PR author and reviewers to solve problems before the PR is merged, rather than having to follow-up afterwards.

That all said, if it's taking a slot from other jobs, then I agree that it's worth reducing the frequency at which CodeQL is run so we can speed up CI for individual PRs.

Reply:
> It's nice to get feedback early while the PR is still being reviewed as this allows the PR author and reviewers to solve problems before the PR is merged, rather than having to follow-up afterwards.

I can't review all pull requests when they are opened. Sometimes we find things after they are merged, while looking for something else. I just wanted to discuss these concerns in the context of the code. In any case, whenever I'm raising concerns and we can agree, I'm sending the fixes myself.